test/createTest.m

```diff
@@ -121,3 +121,28 @@ function createTest()
     fclose(FID);
 
 end
+
+function [result, index] = searchforType(type, graph) 
```

Comment:
The reason I have moved in to inside createTest is that the createTest function is extremely slow if it is creating the hashmap for each json. It took almost 7 minutes on my machine. This is as the test function only searches the graph once per json, as oppose to the actual code which does this several times per json. 

For this reason, I kept the searchForType code but limited it's scope to the createTest function. I can remove it if you would like but the tests are much slower.